### PR TITLE
Enhancement/public privatekey helpers

### PR DESCRIPTION
**Pull request by theflyingcodr**
Added new ToPublicKey & ToPublicKeyWithNet methods to convert an Extended key to an address string.

Also added DeriveChildFromKey method to generate a new ExtendedKey from an existing key using a specified derivationPath.

Removed old Address method and added more tests to cover new methods.

### CODE REVIEW

File: bip32/extendedkey_test.go [hunk 1]
```diff
@@ -648,8 +650,9 @@ func TestExtendedKeyAPI(t *testing.T) {
 			continue
 		}
 
-		addr := key.Address(&chaincfg.MainNet)
-		if addr != test.address {
+		addr, err := key.ToPublicKey()
```
Comment (jadwahab) [inline bip32/extendedkey_test.go:653]: `ToExtendedPublicKey()` and `ToExtendedPrivateKey()`

File: bip32/extendedkey_test.go [hunk 2]
```diff
@@ -821,8 +824,9 @@ func TestZero(t *testing.T) {
 		}
 
 		wantAddr := "1HT7xU2Ngenf7D4yocz2SAcnNLW7rK8d4E"
-		addr := key.Address(&chaincfg.MainNet)
-		if addr != wantAddr {
+		addr, err := key.ToPublicKeyWithNet(true)
```
Comment (jadwahab) [inline bip32/extendedkey_test.go:827]: revert back to `key.Address`

**Comment by jadwahab** [comment]
check https://github.com/btcsuite/btcutil/tree/master/hdkeychain and https://github.com/BitcoinSchema/go-bitcoin/blob/master/hd_key.go

File: bip32/pk.go [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+	return k.ToAddressWithNet(k.IsForNet(&chaincfg.MainNet))
+}
+
+// ToAddressWithNet will return the publicKey bytes for the current extendedkey as well as the address.
+// This takes a mainnet param that if true will return a mainnet address (prefixed with 1), otherwise
+// a testnet address is returned.
+//
+// This method should only be used if you are dealing with a Zeroed ExtendedKey as it
+// cannot determine the network to use. Instead for most instances use ToAddress.
+func (k *ExtendedKey) ToAddressWithNet(mainnet bool) (*PublicKey, error) {
```
Comment (jadwahab) [inline bip32/pk.go:40]: did we get his function from?

Comment (theflyingcodr) [reply bip32/pk.go:40]: Must have been part of the lib already, I've just ported to a different file I think...

Comment (theflyingcodr) [reply bip32/pk.go:40]: It used to be `key.Address(&chaincfg.MainNet)`

**Pull request closed without merging**